### PR TITLE
CLOUDP-67654 dbusers iam auth

### PR DESCRIPTION
**Pull request by robcarlan-mlab**
## Proposed changes

_Jira ticket:_ CLOUDP-67654

Closes #[issue number]

## Checklist

- [] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [] I have added tests that prove my fix is effective or that my feature works
- [] I have added any necessary documentation (if appropriate)
- [] I have run `make fmt` and formatted my code

## Further comments

Happy path:
```
./bin/mongocli atlas dbusers create --awsIAMType USER --x509Type NONE --role atlasAdmin --username "arn:aws:iam::358363220050:user/mongodb-aws-iam-auth-test-user"
{
  "databaseName": "$external",
  "ldapAuthType": "NONE",
  "x509Type": "NONE",
  "awsIAMType": "USER",
  "groupId": "5cac6a2ba6f239298a18a025",
  "roles": [
    {
      "roleName": "atlasAdmin",
      "databaseName": "admin"
    }
  ],
  "username": "arn:aws:iam::358363220050:user/mongodb-aws-iam-auth-test-user"
}
```

Validation:
```
./bin/mongocli atlas dbusers create --awsIAMType ROLE --x509Type NONE --role atlasAdmin --username notanarn
Error: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/5cac6a2ba6f239298a18a025/databaseUsers: 400 (request "Bad Request") The ARN specified in the username field, notanarn, is not valid for AWS IAM type ROLE.

./bin/mongocli atlas dbusers create --awsIAMType ROLE --x509Type MANAGED --role atlasAdmin
Error: cannot supply both AWS IAM and x509 auth

./bin/mongocli atlas dbusers create --awsIAMType INVALID --x509Type NONE --role atlasAdmin --username notanarn
Error: INVALID is an invalid value for awsIAMType. It must be one of: NONE,ROLE,USER
```

### CODE REVIEW

File: internal/cli/atlas/dbusers/create.go [hunk 1]
```diff
@@ -26,17 +26,41 @@ import (
...
+	store      store.DatabaseUserCreator
+}
+
+const (
+	AWSIAMTypeUser   = "USER"
+	AWSIAMTypeRole   = "ROLE"
+	AWSIAMTypeNone   = "NONE"
+	X509TypeManaged  = "MANAGED"
+	X509TypeCustomer = "CUSTOMER"
+	X509TypeNone     = "NONE"
```
Comment (gssbzn) [inline internal/cli/atlas/dbusers/create.go:50]: [non-blocking] I'd vote to just have one None but up to you

File: internal/cli/atlas/dbusers/create.go [hunk 2]
```diff
@@ -26,17 +26,41 @@ import (
...
+	validX509Flags   = []string{X509TypeNone, X509TypeManaged, X509TypeCustomer}
+	validAWSIAMFlags = []string{AWSIAMTypeNone, AWSIAMTypeRole, AWSIAMTypeUser}
+)
+
+func (opts *CreateOpts) isX509Set() bool {
+	return opts.x509Type == X509TypeCustomer || opts.x509Type == X509TypeManaged
+}
+
+func (opts *CreateOpts) isAWSIAMSet() bool {
+	return opts.awsIamType == AWSIAMTypeUser || opts.awsIamType == AWSIAMTypeRole
```
Comment (gssbzn) [inline internal/cli/atlas/dbusers/create.go:63]: similar comment from the x509 PR, it's easier to read just to say 
```suggestion
	return opts.awsIamType != AWSIAMTypeNone
```

File: internal/cli/atlas/dbusers/create.go [hunk 3]
```diff
@@ -76,7 +109,40 @@ func (opts *CreateOpts) Prompt() error {
...
+	if opts.isX509Set() && opts.password != "" {
+		return errors.New("cannot supply both x509 auth and password")
+	}
+
+	if opts.isAWSIAMSet() && opts.password != "" {
+		return errors.New("cannot supply both AWS IAM auth and password")
+	}
+
+	if opts.isAWSIAMSet() && opts.isX509Set() {
+		return errors.New("cannot supply both AWS IAM and x509 auth")
```
Comment (gssbzn) [inline internal/cli/atlas/dbusers/create.go:134]: 💯

**Comment by robcarlan-mlab** [comment]
needed to swap == for !=

**Pull request merged**